File: lib/domain/cultivation_plan/dtos/private_plan_initialize_from_selection_failure.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Dtos
      # Presenter が Symbol ステータスで render_response するための失敗ペイロード
      class PrivatePlanInitializeFromSelectionFailure
        attr_reader :http_status, :message

        def initialize(http_status:, message:)
          @http_status = http_status
          @message = message
        end
      end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/dtos/private_plan_initialize_from_selection_failure.rs
//! Ruby: `Domain::CultivationPlan::Dtos::PrivatePlanInitializeFromSelectionFailure`

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct PrivatePlanInitializeFromSelectionFailure {
    pub http_status: &'static str,
    pub message: String,
}

impl PrivatePlanInitializeFromSelectionFailure {
    pub const HTTP_NOT_FOUND: &'static str = "not_found";
    pub const HTTP_UNPROCESSABLE_ENTITY: &'static str = "unprocessable_entity";

    pub fn new(http_status: &'static str, message: impl Into<String>) -> Self {
        Self {
            http_status,
            message: message.into(),
        }
    }
}
